### PR TITLE
better coverage of sqlalchemy fields

### PR DESCRIPTION
**Pull request by lsbardel**

### CODE REVIEW

File: openapi/data/fields.py [hunk 1]
```diff
@@ -145,6 +153,26 @@ def openapi(self, prop):
         pass
 
 
+@dataclass
+class StrValidator(Validator):
+    min_length: int = 0
+    max_length: int = 0
+
+    def __call__(self, field, value, data=None):
+        value = str(value)
```
Comment (ConorStevenson) [inline openapi/data/fields.py]: Won't this just `repr` anything that doesn't define `__str__`?
`'<__main__.MyClass object at 0x10fbf7630>'` might be the right length...

File: openapi/data/db.py [hunk 2]
```diff
@@ -66,6 +71,14 @@ def en(col):
     )
 
 
+@converter(sa.JSON)
+def js(col):
```
[the remark this answers is not in the document]
Comment (ConorStevenson) [reply openapi/data/db.py]: Ah, beat me to it.

File: openapi/data/fields.py (hunk 1 above)
Comment (markharley) [reply openapi/data/fields.py]: Could do `if isinstance(value, str)`. Do we want to support byte strings with an encoding?

File: openapi/data/db.py [hunk 3]
```diff
@@ -66,6 +71,14 @@ def en(col):
...
+    return (
+        typing.Dict,
+        fields.json_field(**info(col))
+    )
+
+
 @converter(UUIDType)
 def uiid(col):
```
Comment (markharley) [inline openapi/data/db.py]: typo

File: openapi/db/path.py [hunk 4]
```diff
@@ -79,7 +79,7 @@ def db_table(self):
                 **self.api_response_data({'message': 'Invalid JSON payload'})
             )
         data = [self.insert_data(d, body_schema=body_schema) for d in data]
-        values = await self.db.db_insert(table, data)
+        values = await self.db.db_insert(table, data, conn=conn)
```
Comment (ConorStevenson) [inline openapi/db/path.py]: Fixed in my PR first ;)

**Pull request merged**